client/src/pages/Dashboard/components/ConversationList.js

```diff
@@ -0,0 +1,131 @@
+    const [anchorEl, setAnchorEl] = useState(null);
+    const auth = useAuth();
+    const parsedConversations = useMemo(() => {
+        return conversations?.map(conversation => {
+            const otherUser = conversation.users?.filter(user => user.id !== auth.user.id)[0];
+            conversation.otherUser = otherUser;
+            if (conversation.latestMessage)
+                conversation.latestMessage.isRecieved = conversation.latestMessage.fromUserId === auth?.user?.userId;
+            return conversation;
+        });
+    }, [conversations, auth]);
+
+    const handleOpenMenu = (event) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const handleCloseMenu = () => {
+        setAnchorEl(null);
+    };
+
+    const handleSignout = () => {
+        handleCloseMenu();
+        signoutCallback();
+    }
+
+    return (
+        <Grid container className={classes.root} spacing={0} direction='column' justify="flex-start" wrap="nowrap">
+            <Grid className={classes.profile} item >
+                <Grid container direction="row" justify="space-between" alignItems="center" wrap='nowrap'>
+                    <Grid item ><ConversationItem title={auth?.user?.username} onlineStatus={OnlineStatus.ONLINE} interactive={false} /></Grid>
+                    <Grid item>
+                        <IconButton onClick={handleOpenMenu} className={`${classes.icon} ${classes.seeMoreButton}`}>
+                            <MoreHorizIcon />
+                        </IconButton>
+                        <Menu
+                            id="menu"
+                            anchorEl={anchorEl}
+                            keepMounted
+                            open={Boolean(anchorEl)}
+                            onClose={handleCloseMenu}
+                        >
+                            <MenuItem className={classes.icon} onClick={handleSignout}><ExitToAppIcon /> Logout</MenuItem>
+                        </Menu>
+                    </Grid>
+                </Grid>
+            </Grid>
+
+            <Grid item container className={classes.chatHeader} justify="space-between" alignItems="center" >
+                <Typography variant="h3">
+                    Chats
+                </Typography>
+                <Button variant="outlined" color="primary"
+                    startIcon={<AddIcon />}
+                    onClick={addConversationCallback}
+                >
+                    Start conversation
+                </Button>
+            </Grid>
+
+            <Grid item>
+                <SearchInput />
+            </Grid>
+
+            <Grid className={classes.listContainer} item>
+                <Grid className={classes.list} container direction="column" spacing={5}>
+                    {parsedConversations?.map((conversation, i) => {
+                        return (
+                            <Grid onClick={()=>selectConversationCallback(i)} key={conversation.id} item >
```

Comment:
Selecting the conversation based on the index will cause issues down the line if you decide to implement filtering, since the filtered indices won't match up with the indices of the original conversation list. It's recommended to use `id` for this purpose instead.

Reply:
fixed